pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -231,60 +233,15 @@ public class ModelInterpreter implements Serializable {
         }
     }
 
-    /*
-    TODO: The agent handling stuff here is just waiting for step-in-Groovy support..
-     */
-    def nodeOrDockerOrNone(Agent agent, Closure body) {
-        if (agent != null && agent.hasAgent()) {
-            return {
-                nodeWithLabelOrWithout(agent) {
-                    dockerOrWithout(agent, body).call()
-                }.call()
-            }
-        } else {
+    def inDeclarativeAgent(Agent agent, Closure body) {
+        if (agent == null) {
             return {
                 body.call()
             }
-        }
-    }
-
-    def dockerOrWithout(Agent agent, Closure body) {
-        if (agent.docker != null) {
-            return {
-                script.getProperty("docker").image(agent.docker).inside(agent.dockerArgs, {
-                    body.call()
-                })
-            }
         } else {
-            return {
+            return agent.getDeclarativeAgent().getScript(script).run {
```

Comment:
Yup - the overwhelming majority of tests are using `CpsScmFlowDefinition`, which has sandbox on. And I'm adding a change to make sure the ones using `CpsFlowDefinition` turn on sandbox too.